OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -179,7 +180,9 @@ public override Activity Tick(Actor self)
 			if (firstFacing != mobile.Facing)
 			{
 				path.Add(nextCell.Value.First);
-				return ActivityUtils.SequenceActivities(new Turn(self, firstFacing), this);
+				var turnActivity = new Turn(self, firstFacing);
+				turnActivity.NextActivity = this;
+				return turnActivity;
```

Comment:
SequenceActivities already does that. You shouldn't be manipulating NextActivity directly.